assets/js/index.js

```diff
@@ -435,74 +435,29 @@ twentytwenty.modalMenu = {
 		} );
 	},
 
-	// If the current menu item is the last one, return to close button when tab
-	goBackToCloseButton: function() {
+	keepFocusInModal: function() {
 		document.addEventListener( 'keydown', function( event ) {
-			var closeMenuButton, mobileMenu, isDesktop, menuLinks, firstLevelmenuLinks, lastMenuLinkToggleButton, lastToogleSubMenuLinkNotOpened, lastMenuLinkHasSubClosedMenu, socialLinks, hasSocialMenu, lastModalMenuItems, focusedElementParentLi, focusedElementIsInsideModal, lastMenuItem, isFirstModalItem, isLastModalItem;
+			var clickedEl = twentytwenty.toggles.clickedEl,
+				modal, elements, activeEl, lastEl, firstEl;
 
-			closeMenuButton = document.querySelector( '.toggle.close-nav-toggle' );
-			mobileMenu = document.querySelector( '.mobile-menu' );
+			if ( clickedEl ) {
+				modal = document.querySelector( clickedEl.dataset.toggleTarget );
 
-			if ( mobileMenu ) {
-				return false;
-			}
-
-			isDesktop = window.getComputedStyle( mobileMenu, null ).getPropertyValue( 'display' ) === 'none';
-
-			menuLinks = isDesktop ?
-				document.querySelectorAll( '.menu-modal .expanded-menu .modal-menu li' ) :
-				document.querySelectorAll( '.menu-modal .mobile-menu .modal-menu li' );
-
-			firstLevelmenuLinks = isDesktop ?
-				document.querySelectorAll( '.menu-modal .expanded-menu .modal-menu > li' ) :
-				document.querySelectorAll( '.menu-modal .mobile-menu .modal-menu > li' );
-
-			if ( firstLevelmenuLinks ) {
-				return false;
-			}
-
-			lastMenuLinkToggleButton = firstLevelmenuLinks[firstLevelmenuLinks.length - 1].querySelector( '.sub-menu-toggle' ) || undefined;
-			lastMenuLinkHasSubClosedMenu = lastMenuLinkToggleButton && ! lastMenuLinkToggleButton.classList.contains( 'active' );
-
-			lastToogleSubMenuLinkNotOpened = isDesktop ?
-				document.querySelector( '.menu-modal .expanded-menu .modal-menu .sub-menu .sub-menu-toggle:not(.active)' ) :
-				document.querySelector( '.menu-modal .mobile-menu .sub-menu .sub-menu-toggle:not(.active)' );
-
-			socialLinks = document.querySelectorAll( '.menu-modal .social-menu li' );
-			hasSocialMenu = document.querySelectorAll( '.menu-modal .social-menu' ).length > 0;
-			lastModalMenuItems = hasSocialMenu ? socialLinks : menuLinks;
-			focusedElementParentLi = twentytwentyFindParents( event.target, 'li' );
-			focusedElementIsInsideModal = twentytwentyFindParents( event.target, '.menu-modal' ).length > 0;
+				elements = modal.querySelectorAll( 'input, a, button' );
+				elements = Array.prototype.slice.call( elements );
 
-			lastMenuItem = lastModalMenuItems[lastModalMenuItems.length - 1];
+				lastEl = elements[ elements.length - 1 ];
+				firstEl = elements[0];
+				activeEl = document.activeElement;
```

Comment:
Are these variables taking into account that the last menu item can have submenus open or closed?

Reply:
I hope so 😆

Reply:
You're right about that, I'll have to fix it 😄

Reply:
Fixed in 6a9fec2

Please check it if you have the time 😄

Reply:
You are making a huuuge refactor 😆 , let you know if fails.

Reply:
Works like a charm :-)

Reply:
thanks 😄 marking this as resolved